tests/team.c

```diff
@@ -0,0 +1,85 @@
+/* -*- Mode: C; c-basic-offset:4 ; -*- */
+/*
+ * (C) 2018 by Argonne National Laboratory.
+ *     See COPYRIGHT in top-level directory.
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <shmem.h>
+#include <shmemx.h>
+
+#define NSPACES 10
+
+int main(int argc, char *argv[])
+{
+    int mype, i, total_pes;
+    int num_err = 0;
+
+    shmem_init();
+
+    total_pes = shmem_n_pes();
+    mype = shmem_my_pe();
+
+    if (total_pes < 5) {
+        shmem_finalize();
+        fprintf(stdout, "Need at least 5 processes\n");
+        return 0;
+    }
```

Comment:
I don't see why at least 5 processes is required. It looks like all the tests will work with even only 1 PE.  Can you please explain it for me?

Reply:
This is mainly because of the `SHMEM_team_split_2d` which does a X,Y split teams on a 2D grid. 4 processes will have a 2x2 grid, but it does not cover the case where the last row is not full. So, 5 processes are needed for the grid to be 3x2 with last row only have one process.

Reply:
I see. But will the program work if with less than 5 processes? I mean if the program will work fine, we don't need stop the test at beginning if np<5. We want to set np=5 in our test script to cover the 3x2 case, but if the user uses `-np 4` for example, the program can still test 2x2 case, right? :-)

Reply:
Yes. that's right. In fact, 2 processes should also work. It just not exercising the code to the full extent.

We probably want to expand the capability of multiprocess testing of the test infrastructure in the future.

Reply:
I agree. Would you mind preparing a ticket for it?

Reply:
Meanwhile, I think it is safe to always perform the test when `np<5`. Do you think we can delete the early exit check?

Reply:
@yfguo Looks like here is the last unresolved issue. Can you please take a look at it so that we can merge it?

Reply:
right away.

Reply:
This is done.

Reply:
FYI: opened a ticket for the test framework improve #110